rust/src/treefile.rs

```diff
@@ -612,6 +616,13 @@ struct Rojig {
     description: Option<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+enum Include {
+    Single(String),
+    Multiple(Vec<String>)
```

Comment:
Are duplicate entries meant to be valid in the vector here?

Reply:
I can't think of a valid use case to do so, but it also in most cases wouldn't break anything.  Tried to make it a `BTreeSet` but serde doesn't make it an error for the source list to have duplicates.

Reply:
Trying to dedupe without doing a `realpath` first wouldn't be super effective anyway. I agree it's not inherently wrong though, it's just a funny thing to do.

Reply:
We could do the standard thing of building up a `Set<(device, inode)>` as we traverse.  Yeah...it's probably worth it, having duplicate includes silently could result in confusing things like running a postprocess script twice (or more).

Reply:
Sorry for kicking this hornets' nest, I was just curious about the high-level de-dup semantics.

On a similar note, I think that the current `BTree*` structure will re-order according to sorting order, not to input/insertion order. I don't know if that is on purpose as well.

Reply:
That's a good point. This matters if there's ordering dependencies between non-declarative parts (which is basically just the postprocess scripts).

Reply:
Hmm.  I guess it's probably better to use `Vec` indeed.